### ISSUE TITLE
auto version: Confusing behavior of Conventional Commits Plugin

### ISSUE THREAD

**Issue by Yeti-or** [intuit/auto#1719]
**Describe the bug**

First of all we don't use labels on our PR, and use `"conventional-commits", "npm" & "slack"` plugins.
We have two problems with CC Plugin, and I believe they are connected.

Let's discuss master branch flow:

When you merge PR and don't label PR;
cc-plugin will omit merge commit from commits that participate in `auto version`:

https://github.com/intuit/auto/blob/8b4d1f23e522ccf5695b1a4342b5c5d325522604/plugins/conventional-commits/src/index.ts#L189-L202

so if you have something like these:

```
*   8d32016 Merge pull request #3 from yeti-or.test-3
|\
| * d7c4936  feat: new shinny feature
|/
*
```
Everything is okay, and `auto version` returns `minor`

For case when we have no need to release
```
*   8d32016 Merge pull request #3 from yeti-or.test-3
|\
| * d7c4936  build: some chore work
|/
*
```
`auto version` will return nothing. 

But if we have two commits:

```
*   8d32016 Merge pull request #3 from yeti-or.test-3
|\
| * d7c4936  build: some chore work
| * dbc91a0  feat: new shinny feature
|/
*
```

I would expect to get `minor` but instead I get nothing from `auto version`
because `auto version` checks necessity of release in first commit. And first one was omited by cc-plugin.

https://github.com/intuit/auto/blob/8b4d1f23e522ccf5695b1a4342b5c5d325522604/packages/core/src/semver.ts#L219-L227

So whole pr with bunch of feat/major commits will be skipped.
And you couldn't predict these because canary version was published okay in your PR.

**Second problem**:

I suppose it's similar problem. If we write in pull-request summary something using conventional-commit msg:

```
Very useful PR: several features here
```

Title of pr will be provided to merge commit.

https://github.com/intuit/auto/blob/8b4d1f23e522ccf5695b1a4342b5c5d325522604/packages/core/src/release.ts#L524

Then CC-Plugin will try to determine version of these commit:

https://github.com/intuit/auto/blob/8b4d1f23e522ccf5695b1a4342b5c5d325522604/plugins/conventional-commits/src/index.ts#L145

inside `getBump`  CC- Plugin will parse PR title.
https://github.com/intuit/auto/blob/8b4d1f23e522ccf5695b1a4342b5c5d325522604/plugins/conventional-commits/src/index.ts#L112

conventionalCommit would be smth like this:

```
< {
<   type: 'Very useful PR',
<   scope: null,
<   subject: 'several features here',
<   merge: null,
<   header: 'several features here',
<   body: 'undefined',
<   footer: null,
<   notes: [],
<   references: [],
<   mentions: [],
<   revert: null
< }
```

so `label` would be `'skip'`

And moreover this commit wouldn't be omitted, because it has label and cc-plugin omits only commits without labels:

https://github.com/intuit/auto/blob/8b4d1f23e522ccf5695b1a4342b5c5d325522604/plugins/conventional-commits/src/index.ts#L182

So you could skip release by using conventional commits in your pr title. But I didn't expected it, and [docs]( https://intuit.github.io/auto/docs/generated/conventional-commits) didn't prepared me.

It could be done in such steps:

* create pr from one chore commit
* github suggest you to use commit message as PR title
* add one more feat/fix commit after review
* merge pr with two commits
* don't get release (skipped because of PR title)

still get canary version in pr though

**Expected behavior**

1) Expect auto to treat pr as a whole bunch of changes, and not to skip release because of first commit, but look at them all and skip only if all commits are `skip-release`
2) Expect CC-Plugin not to use pr title as check for skipping release. At least don't skip on everything that has `someword: text`.

**Environment information:**

```txt
Environment Information:

"auto" version: v10.6.2
"git"  version: v2.29.1
"node" version: v12.16.1

Project Information:

✔ Repository:      ***
✔ Author Name:     Vasiliy Loginevskiy
✔ Author Email:    yeti.or@gmail.com
✖ Current Version: v
✔ Latest Release:  ***
✖ Labels configured on GitHub project (Try running "auto create-labels")

GitHub Token Information:

✔ Token:            [Token starting with ***]
✖ Repo Permission:  read
✔ User:             ***
✔ API:
✔ Enabled Scopes:   gist, notifications, repo, user, workflow, write:discussion, write:packages
✔ Rate Limit:       4997/5000
```

**Comment by hipstersmoothie** [comment]
This is a tricky one to fix!

Just to be clear you expect:

```
*   8d32016 Merge pull request #3 from yeti-or.test-3 (PR TItle: Enhancement PR)
|\
| * d7c4936  build: some chore work
| * dbc91a0  feat: new shinny feature
|/
*
```

to create a minor release and for the changelog to look something like

```md
#### 🚀 Enhancement

- Enhancement PR
- new shiny feature

####  Build

- some chore work

#### Authors: 1

- Andrew Lisowski ([@hipstersmoothie](https://github.com/hipstersmoothie))
```

**Comment by hipstersmoothie** [comment]
> Expect auto to treat pr as a whole bunch of changes, and not to skip release because of first commit, but look at them all and skip only if all commits are skip-release

The above comments pertain to this and you can test out if I fixed it by installing the canary version from [this PR](https://github.com/intuit/auto/pull/1723).

> Expect CC-Plugin not to use pr title as check for skipping release. At least don't skip on everything that has `someword: text`

This breaks with how `auto` has worked so far. We treat the PR title as a commit message. If that title has a conventional commit that isnt a patch, minor, major it defaults to skip. If you add a label to the PR though that will take precedence.

**Comment by aleclarson** [comment]
It would be great if #1723 checked all commits after the latest version, instead of only the most recent commit.

**Comment by adierkens** [comment]
:rocket: Issue was released in [`v10.23.0`](https://github.com/intuit/auto/releases/tag/v10.23.0) :rocket:

**Comment by hipstersmoothie** [comment]
@Yeti-or if you find any remaining issue with the plugin feel free to open another issue. Thanks for the in depth bug report!
